apps/gdalmdimtranslate_lib.cpp

```diff
@@ -55,6 +56,122 @@ struct GDALMultiDimTranslateOptions
     bool bUpdate = false;
 };
 
+/*************************************************************************/
+/*             GDALMultiDimTranslateAppOptionsGetParser()                */
+/************************************************************************/
+
+static std::unique_ptr<GDALArgumentParser>
+GDALMultiDimTranslateAppOptionsGetParser(
+    GDALMultiDimTranslateOptions *psOptions,
+    GDALMultiDimTranslateOptionsForBinary *psOptionsForBinary)
+{
+    auto argParser = std::make_unique<GDALArgumentParser>(
+        "gdalmdimtranslate", /* bForBinary=*/psOptionsForBinary != nullptr);
+
+    argParser->add_description(
+        _("Converts multidimensional data between different formats, and "
+          "performs subsetting."));
+
+    argParser->add_epilog(
+        _("For more details, consult "
+          "https://gdal.org/programs/gdalmdimtranslate.html"));
+
+    if (psOptionsForBinary)
+    {
+        argParser->add_input_format_argument(
+            &psOptionsForBinary->aosAllowInputDrivers);
+    }
+
+    argParser->add_output_format_argument(psOptions->osFormat);
+
+    argParser->add_creation_options_argument(psOptions->aosCreateOptions);
+
+    auto &group = argParser->add_mutually_exclusive_group();
+    group.add_argument("-array")
+        .metavar("<array_spec")
+        .append()
+        .store_into(psOptions->aosArraySpec)
+        .help(_(
+            "Select a single array instead of converting the whole dataset."));
+
+    argParser->add_argument("-arrayoption")
+        .metavar("<NAME>=<VALUE>")
+        .append()
+        .action([psOptions](const std::string &s)
+                { psOptions->aosArrayOptions.AddString(s.c_str()); })
+        .help(_("Option passed to GDALGroup::GetMDArrayNames() to filter "
+                "reported arrays."));
+
+    group.add_argument("-group")
+        .metavar("<group_spec>")
+        .append()
+        .store_into(psOptions->aosGroup)
+        .help(_(
+            "Select a single group instead of converting the whole dataset."));
+
+    // Note: this is mutually exclusive with "view" option in -array
+    argParser->add_argument("-subset")
+        .metavar("<subset_spec>")
+        .append()
+        .store_into(psOptions->aosSubset)
+        .help(_("Select a subset of the data."));
+
+    // Note: this is mutually exclusive with "view" option in -array
+    argParser->add_argument("-scaleaxes")
+        .metavar("<scaleaxes_spec>")
+        .scan<'i', int>()
```

Comment:
As we support multiple values separated by comma, scan<'i', int> is likely not appropriate
```suggestion
```

Reply:
done